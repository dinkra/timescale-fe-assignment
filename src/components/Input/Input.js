import styled from 'styled-components';
import search from '../../theme/svgs/search';

const Wrapper = styled.div`
  position: relative;
`;

const StyledIcon = styled.div`
  position: absolute;
  top: 10px;
  left: 10px;
`;

const StyledInput = styled.input`
  box-sizing: border-box;
  width: 100%;
  padding: 10px;
  border: solid 1px ${({ theme }) => theme.navy500};
  border-radius: 2px;
  background-color: ${({ theme }) => theme.white};
  outline: none;
  color: ${({ theme }) => theme.black};
  font-size: ${({ theme }) => theme.fontSizeSmall};
  text-overflow: ellipsis;
  cursor: text;

  ::placeholder {
    color: ${(p) => p.theme.navy600};
  }

  &:focus {
    transition: 0.2s linear;
    border-color: ${(p) => p.theme.navy600};
    box-shadow: 0 0 2px ${(p) => p.theme.navy600} inset;
  }

  ${({ hasIcon }) => hasIcon && 'padding-left: 34px;'}
`;

const Input = (props) => (
  <Wrapper>
    {props.hasIcon && <StyledIcon>{search}</StyledIcon>}
    <StyledInput {...props} />
  </Wrapper>
);

export default Input;
