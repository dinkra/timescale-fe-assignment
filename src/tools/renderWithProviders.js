import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

// override render method
export { customRender as render };
