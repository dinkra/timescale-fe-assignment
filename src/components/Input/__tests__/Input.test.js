import { render, fireEvent } from '../../../tools/renderWithProviders';
import Input from '../Input';

describe('Input', () => {
  it('should match snapshot', () => {
    const wrapper = render(<Input placeholder="Type something here" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with icon', () => {
    const wrapper = render(<Input placeholder="Type something here" hasIcon />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should listen change events', () => {
    const onChangeMock = jest.fn();

    const { getByDisplayValue } = render(
      <Input
        id="name"
        placeholder="Type something here"
        onChange={onChangeMock}
        value="value"
      />
    );
    const input = getByDisplayValue(/value/i);
    fireEvent.change(input, { target: { value: 'new value' } });

    expect(onChangeMock).toBeCalled();
  });
});
