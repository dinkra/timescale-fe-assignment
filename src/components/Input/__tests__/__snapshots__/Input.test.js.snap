// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM jaBgCn"
      >
        <input
          class="sc-dlnjwi gQNjWJ"
          placeholder="Type something here"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM jaBgCn"
    >
      <input
        class="sc-dlnjwi gQNjWJ"
        placeholder="Type something here"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input should match snapshot with icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM jaBgCn"
      >
        <div
          class="sc-gtsrHT gtPaqp"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.16666 0.666656C3.12909 0.666656 0.666656 3.12909 0.666656 6.16666C0.666656 9.20422 3.12909 11.6667 6.16666 11.6667C7.5054 11.6667 8.73244 11.1883 9.68618 10.3933L13.4798 14.1869C13.675 14.3821 13.9916 14.3821 14.1869 14.1869C14.3821 13.9916 14.3821 13.675 14.1869 13.4798L10.3933 9.68618C11.1883 8.73244 11.6667 7.5054 11.6667 6.16666C11.6667 3.12909 9.20422 0.666656 6.16666 0.666656ZM1.66666 6.16666C1.66666 3.68138 3.68138 1.66666 6.16666 1.66666C8.65194 1.66666 10.6667 3.68138 10.6667 6.16666C10.6667 8.65194 8.65194 10.6667 6.16666 10.6667C3.68138 10.6667 1.66666 8.65194 1.66666 6.16666Z"
              fill="#898E9A"
            />
          </svg>
        </div>
        <input
          class="sc-dlnjwi cVaKhU"
          placeholder="Type something here"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM jaBgCn"
    >
      <div
        class="sc-gtsrHT gtPaqp"
      >
        <svg
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.16666 0.666656C3.12909 0.666656 0.666656 3.12909 0.666656 6.16666C0.666656 9.20422 3.12909 11.6667 6.16666 11.6667C7.5054 11.6667 8.73244 11.1883 9.68618 10.3933L13.4798 14.1869C13.675 14.3821 13.9916 14.3821 14.1869 14.1869C14.3821 13.9916 14.3821 13.675 14.1869 13.4798L10.3933 9.68618C11.1883 8.73244 11.6667 7.5054 11.6667 6.16666C11.6667 3.12909 9.20422 0.666656 6.16666 0.666656ZM1.66666 6.16666C1.66666 3.68138 3.68138 1.66666 6.16666 1.66666C8.65194 1.66666 10.6667 3.68138 10.6667 6.16666C10.6667 8.65194 8.65194 10.6667 6.16666 10.6667C3.68138 10.6667 1.66666 8.65194 1.66666 6.16666Z"
            fill="#898E9A"
          />
        </svg>
      </div>
      <input
        class="sc-dlnjwi cVaKhU"
        placeholder="Type something here"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
